src/main/java/Duke.java

```diff
@@ -123,53 +28,108 @@ public static String promptEntry() {
        return sc.next();
     }
 
-    public static void printList(ArrayList<Task> toDoList) {
-        int n = 1;
+    public void run() {
+
+        this.ui.welcome();
 
-        if(toDoList.isEmpty()){
+        String command = promptEntry();
```

Comment:
Further OOP can be achieved with the processing of commands shifted to a command class. Putting everything here in the run method() dissolves the purpose of OOP